### PR TITLE
Added target framework .NET 6 and updated dependencies

### PR DESCRIPTION
**Pull request by Havunen**
Fixes https://github.com/Tornhoof/SpanJson/issues/158

### CODE REVIEW

File: SpanJson.AspNetCore.Formatter.Tests/SpanJson.AspNetCore.Formatter.Tests.csproj [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-    <TargetFrameworks>netcoreapp3.1;net5.0</TargetFrameworks>
+    <TargetFrameworks>netcoreapp3.1;net5.0;net6.0</TargetFrameworks>
     <IsPackable>false</IsPackable>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="3.1.9" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.8.0" />
-    <PackageReference Include="Moq" Version="4.15.1" />
+    <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="3.1.21" />
```
Comment (Havunen) [inline SpanJson.AspNetCore.Formatter.Tests/SpanJson.AspNetCore.Formatter.Tests.csproj:9]: to have latest test host supported, we would need to split this dependency based on target framework version

**Review by silkfire — COMMENTED** [review]
While you're at it, maybe you could also remove the dependency on the `System.Runtime.CompilerServices.Unsafe` NuGet package, as `Unsafe` is already part of the framework since .NET Core 3.1?

**Comment by Tornhoof** [comment]
Thank you for this PR, the azure devops yaml changes need a bit of work. I have a different branch locally with the required changes for .net 6. And I need to get the integration in github working again, apparently something broke there.

**Comment by Tornhoof** [comment]
Ok the integration works again, the changed azure devops yaml did the trick. I'll merge this now.

**Pull request merged**